bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java

```diff
@@ -81,20 +113,101 @@
 
   private final String host;
 
+  private final HeapSizeManager heapSizeManager;
   private final AsyncExecutor asyncExecutor;
+  private final ExecutorService executorService;
+  private final BigtableOptions options;
+
+  private final LinkedBlockingQueue<MutationOperation> mutationsToBeSent =
+      new LinkedBlockingQueue<>();
+
+  /**
+   * A counter for the number of {@link #mutationWorker}s are
+   * active.
+   */
+  private final AtomicInteger activeMutationWorkers = new AtomicInteger();
+
+  /**
+   * This {@link Runnable} pulls a mutation from {@link #mutationsToBeSent}, and calls {{@link
+   * #issueRequest(Mutation, long)}.
+   */
+  private final Runnable mutationWorker = new Runnable() {
+    @Override
+    public void run() {
+      activeMutationWorkers.incrementAndGet();
+      try {
+        while (!closed) {
+          try {
+            MutationOperation operation =
+                mutationsToBeSent.poll(MUTATION_TO_BE_SENT_WAIT_MS, TimeUnit.MILLISECONDS);
+            // The operation can be null if a timeout occurs.
+            if (operation != null) {
+              if (operation.isCloseMarker) {
+                mutationsToBeSent.add(operation);
```

Comment:
I'm still not convinced that a CLOSE_MARKER is needed. Once mutator is closed no more mutations may be added. If the mutation_worker's main loop is changed to the following it seems to me that it would work
while (!closed || !mutationsToSend.isEmpty() ) {
   try {
     poll and send
   }  catch (InterruptedException e) {
     Thread.interrupted();
     // do not break;
   } catch (Exception e) {
    // as before
  } 
}